core/templates/dev/head/pages/exploration-player-page/services/audio-translation-language.service.spec.ts

```diff
@@ -25,72 +25,206 @@ describe('Audio translation language service', () => {
   let atls: AudioTranslationLanguageService;
   beforeEach(() => {
     atls = TestBed.get(AudioTranslationLanguageService);
+
+    spyOn(window.speechSynthesis, 'getVoices').and.returnValue([{
+      'default': false,
+      lang: 'en-GB',
+      localService: false,
+      name: 'UK English',
+      voiceURI: 'UK English'
+    }]);
   });
 
   it('should properly initialize the current audio language when ' +
      'a preferred language is set', () => {
     let allAudioLanguageCodesInExploration = ['hi-en', 'en'];
     let preferredLanguageCode = 'hi-en';
     let explorationLanguageCode = 'hi';
+    const automaticTextToSpeechEnabled = false;
+
     atls.init(allAudioLanguageCodesInExploration, preferredLanguageCode,
-      explorationLanguageCode, false);
+      explorationLanguageCode, automaticTextToSpeechEnabled);
+    expect(atls.isAutomaticTextToSpeechEnabled()).toBe(
+      automaticTextToSpeechEnabled);
     expect(atls.getCurrentAudioLanguageCode()).toEqual('hi-en');
+    expect(atls.getallAudioLanguageCodesInExploration()).toBe(
```

Comment:
```suggestion
    expect(atls.getAllAudioLanguageCodesInExploration()).toBe(
```

Can you also edit the original function name?

Reply:
Done.